### ISSUE TITLE
Exporting and Importing Tasks

### ISSUE THREAD

**Issue by gsupreeth** [getting-things-gnome/gtg#632]
It would be nice to have the ability to export tasks into either a single plain text file, or a folder of plain text file. Import would be nice as well.

The case for this is that I've had to retype my tasks a couple of times when GTG froze and I panicked. 

For context, I used to use todo.txt before I noticed GTG has been resurrected, and it was nice to have a malleable format. GTG is of course far superior because of the ability to nest tasks and type out long-form descriptions and notes. This ability of GTG to capture ideas in the long-form text makes it important to be able to get it out and into other text processors if I need to. I currently cut and paste or retype.

**Comment by diegogangl** [comment]
This would be a good way to kickstart the new exporter plugin. In the mean time, note that all the data is stored in a file called `gtg_data.xml` and you can edit/copy/paste with any text editor. There's also a backups folder. Backups are made at every change + daily.

**Comment by bepolymathe** [comment]
Yes it seems to me that it would be very interesting to have clean exports for gtg...

**Comment by osslate** [comment]
Would be interested in working on this functionality.

What format makes sense for this plugin? I've personally enjoyed Todoist's CSV-based import/export system; sometimes quicker to create a spreadsheet of tasks in Todoist Template format where the manipulation tools in `insert spreadsheet app` make sense.

Either way, I would also love to add functionality to the plugin to handle Todoist template CSVs too — would be handy to export tasks from Todoist and import them straight into GTG.

**Comment by bepolymathe** [comment]
A markdown export ?

**Comment by diegogangl** [comment]
> What format makes sense for this plugin?

GTG already stores everything in a XML file, so a good first feature would be merging data files. This should be easy since all the parsing/serializing is already done and you can copy from there. The only problem is collisions between tag names and (rarely) UUIDs. These would have to be solved in some way when importing.

The second "first feature" would be markdown export. This is a first step for all the other formats (html, pdf, latex), because once we have markdown we can use pandoc to convert to those formats. I don't think we can do markdown import since the metadata would be limited.

CSV could also be really useful, since it's such an easy language to parse and use.

**Comment by nekohayo** [comment]
The problem I could imagine with markdown, CSV, etc. (basically anything that isn't GTG's XML format) is... how are you not going to lose data integrity, such as start/due dates, tags, and related (sub/parent) tasks?

And why do you want file exporting?
* If that is only for backups reasons... well, GTG already does backups (multiple times per day, and keeps a back up of every day too) within its data folder... why duplicate this outside that folder, essentially?
* If the goal is interoperability with Todoist, wouldn't it be a better user experience to have a sync backend for it? I'd be happy to see folks working on fixing things in https://github.com/getting-things-gnome/gtg/tree/master/GTG/backends/unmaintained and/or using that as a base for a todoist sync backend...
